src/meta/src/rpc/service/cloud_service.rs

```diff
@@ -0,0 +1,190 @@
+                            .is_vpc_endpoint_ready(service.endpoint_id.as_str())
+                            .await;
+                        match privatelink_status {
+                            Err(e) => {
+                                return Ok(new_rwc_validate_fail_response(
+                                    ErrorType::PrivatelinkUnavailable,
+                                    e.to_string(),
+                                ));
+                            }
+                            Ok(false) => {
+                                return Ok(new_rwc_validate_fail_response(
+                                    ErrorType::PrivatelinkUnavailable,
+                                    format!(
+                                        "Private link endpoint {} is not ready",
+                                        service.endpoint_id,
+                                    ),
+                                ));
+                            }
+                            _ => (),
+                        };
+                        if let Err(e) =
+                            insert_privatelink_broker_rewrite_map(&service, &mut source_cfg)
+                        {
+                            return Ok(new_rwc_validate_fail_response(
+                                ErrorType::PrivatelinkResolveErr,
+                                e.to_string(),
+                            ));
+                        }
+                    }
+                }
+            } else {
+                return Ok(new_rwc_validate_fail_response(
+                    ErrorType::PrivatelinkResolveErr,
+                    format!("connection {} has no info available", connection_id),
+                ));
+            }
+        }
+        // try fetch kafka metadata, return error message on failure
+        let source_cfg: HashMap<String, String> =
+            source_cfg.into_iter().map(|(k, v)| (k, v)).collect();
+        let props = ConnectorProperties::extract(source_cfg);
+        if let Err(e) = props {
+            return Ok(new_rwc_validate_fail_response(
+                ErrorType::KafkaInvalidProperties,
+                e.to_string(),
+            ));
+        };
+        let enumerator =
+            SplitEnumeratorImpl::create(props.unwrap(), SourceEnumeratorContext::default().into())
+                .await;
+        if let Err(e) = enumerator {
+            return Ok(new_rwc_validate_fail_response(
+                ErrorType::KafkaInvalidProperties,
+                e.to_string(),
+            ));
+        }
+        if let Err(e) = enumerator.unwrap().list_splits().await {
+            let error_message = e.to_string();
+            if error_message.contains("BrokerTransportFailure") {
+                return Ok(new_rwc_validate_fail_response(
+                    ErrorType::KafkaBrokerUnreachable,
+                    e.to_string(),
+                ));
+            }
+            static TOPIC_NOT_FOUND: LazyLock<Regex> =
```

Comment:
Why define the `Regex` in the function?

Reply:
since we want to return the type of error, we need to use regex to match the error message returned by Kafka (topic * not found).

Reply:
we only have one error regex?

Reply:
currently in my tests, this is the only error that needs a regex..
for other types of errors, the user may refer to the error message, which will be printed to stderr.